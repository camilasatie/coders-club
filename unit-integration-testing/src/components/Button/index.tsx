interface IButtonProps {
  label: string;
  onClick?: () => void;
}

function Button(props: IButtonProps) {
  return (
    <button onClick={props.onClick}>
      {props.label}
    </button>
  )
}

export default Button;
