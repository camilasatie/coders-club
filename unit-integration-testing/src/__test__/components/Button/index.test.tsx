import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Button from '../../../components/Button';

describe('test button successfully', () => {
  it('should render the button', () => {
    const buttonText = 'Login';
    const { container } = render(<Button label={buttonText } />);

    const button = screen.getByRole('button', {name: buttonText})
    
    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })

  it('should call onClick function successfully', () => {
    const buttonText = 'Login';
    const onClick = jest.fn();

    render(<Button label={buttonText } onClick={onClick} />);

    const button = screen.getByRole('button', {name: buttonText})
    
    userEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  })
})