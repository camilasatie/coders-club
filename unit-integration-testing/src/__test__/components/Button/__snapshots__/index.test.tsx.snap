// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button successfully should render the button 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
